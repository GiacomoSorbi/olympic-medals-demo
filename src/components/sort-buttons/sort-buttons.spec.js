import * as React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import SortButtons from './'

describe('SortButtons should', () => {
  it('exist', () => {
    expect(SortButtons).not.toBeUndefined()
  })

  it('have a consistent structure', () => {
    const wrapper = mount(
      <SortButtons
        sortCriteria='silver-desc'
        color='bronze'
        onClick={jest.fn()}
      />,
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
