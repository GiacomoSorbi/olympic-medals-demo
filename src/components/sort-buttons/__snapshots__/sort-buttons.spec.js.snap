// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortButtons should have a consistent structure 1`] = `
<SortButtons
  color="bronze"
  onClick={[MockFunction]}
  sortCriteria="silver-desc"
>
  <div
    className="sort-buttons d-flex flex-column"
  >
    <Button
      color="primary"
      data-sort-direction="bronze-desc"
      onClick={[MockFunction]}
      tag="button"
    >
      <button
        aria-label={null}
        className="btn btn-primary"
        data-sort-direction="bronze-desc"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="edit row"
          src="arrow.svg"
        />
      </button>
    </Button>
    <Button
      className="reversed"
      color="primary"
      data-sort-direction="bronze-asc"
      onClick={[MockFunction]}
      tag="button"
    >
      <button
        aria-label={null}
        className="reversed btn btn-primary"
        data-sort-direction="bronze-asc"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="edit row"
          src="arrow.svg"
        />
      </button>
    </Button>
  </div>
</SortButtons>
`;
